webui/react/src/pages/WorkspaceDetails.tsx

```diff
@@ -256,24 +239,24 @@ const WorkspaceDetails: React.FC = () => {
     return <Spinner spinning tip={`Loading workspace ${workspaceId} details...`} />;
   } else if (isNaN(id)) {
     return <Message title={`Invalid Workspace ID ${workspaceId}`} />;
-  } else if (pageError && !isNotFound(pageError)) {
-    const message = `Unable to fetch Workspace ${workspaceId}`;
-    return <Message title={message} type={MessageType.Warning} />;
-  } else if ((!rbacLoading && !canViewWorkspaceFlag) || (pageError && isNotFound(pageError))) {
+  } else if ((!rbacLoading && !canViewWorkspaceFlag) || workspace === NotLoaded) {
```

Comment:
hmmm this is kind of a stumper...

So without the fetch, we lose info on failures, specifically the page error. With the workspaceStore we ONLY know whether the workspace is loaded or not. We don't know if there was an error and that there was an attempt at fetching. So the logic here is then will always resolve to `<PageNotFound />` when workspace is `NotLoaded`. The bottom `Loadable.match` won't ever call the `NotLoaded` routine because it will always exit here instead... Meaning the spinner will never render below.

Reply:
`getWorkspace(id?: number): Observable<Loadable<Workspace>>` is clearly an incorrect type definition because there's no way there's a worskpace for every number. This should be `getWorkspace(id?: number): Observable<Loadable<Workspace | null>>` since there may be no Workspace for that id.

Reply:
Also why is `id` optional?

Reply:
ok i tried making `id` required, which would make sense
we have ~5 places in the code (including this PR) where we get the ID from something we don't know is loaded yet:
`useObservable(workspaceStore.getWorkspace(modelVersion?.model.workspaceId))`

I could write `modelVersion?.model.workspaceId || 0` and get a `NotLoaded` response, but as I understand the useObservable hook we cannot put it inside of `if (modelVersion) { ... }`

Reply:
> ok i tried making `id` required, which would make sense we have ~5 places in the code (including this PR) where we get the ID from something we don't know is loaded yet: `useObservable(workspaceStore.getWorkspace(modelVersion?.model.workspaceId))`
> 
> I could write `modelVersion?.model.workspaceId || 0` and get a `NotLoaded` response, but as I understand the useObservable hook we cannot put it inside of `if (modelVersion) { ... }`

Yeah this was one of the reasons why `id` was optional when I refactored it.

Reply:
@mapmeld It's perfectly fine to accept a `Loadable<id>`, see [the README](https://github.com/determined-ai/determined/blob/main/webui/react/src/utils/README.md#designing-apis-with-loadable).